src/leap/common/events/server.py

```diff
@@ -24,213 +24,94 @@
-
-    def unregister(self, controller, request, done):
+        Initialize the events server.
         """
-        Unregister a client port so it will not be signaled when specific
-        events come in.
-
-        :param controller: used to mediate a single method call
-        :type controller: protobuf.socketrpc.controller.SocketRpcController
-        :param request: the request received from the client
-        :type request: leap.common.events.events_pb2.RegisterRequest
-        :param done: callback to be called when done
-        :type done: protobuf.socketrpc.server.Callback
-        """
-        logger.info(
-            "Received unregistration request: %s..." % str(request)[:40])
-        # remove client port from signal list
-        response = proto.EventResponse()
-        if request.event in registered_clients:
-            try:
-                registered_clients[request.event].remove(request.port)
-                response.status = proto.EventResponse.OK
-            except KeyError:
-                response.status = proto.EventsResponse.ERROR
-                response.result = 'Port %d not registered.' % request.port
-        # send response back to client
-        logger.debug('sending response back')
-        done.run(response)
-
-    def signal(self, controller, request, done):
+        self._factory = txzmq.ZmqFactory()
+        self._factory.registerForShutdown()
+        self._zmq_bind_pub(pub_port)
+        self._zmq_bind_pull(sig_port)
+
+    def _zmq_bind_pub(self, port):
         """
-        Perform an RPC call to signal all clients registered to receive a
-        specific signal.
-
-        :param controller: used to mediate a single method call
-        :type controller: protobuf.socketrpc.controller.SocketRpcController
-        :param request: the request received from the client
-        :type request: leap.common.events.events_pb2.SignalRequest
-        :param done: callback to be called when done
-        :type done: protobuf.socketrpc.server.Callback
+        Bind a PUB socket to a port.
+
+        :param port: The port to bind to.
+        :type port: int
         """
-        logger.debug('Received signal from client: %s...', str(request)[:40])
-        # send signal to all registered clients
-        # TODO: verify signal auth
-        if request.event in registered_clients:
-            for port in registered_clients[request.event]:
-
-                def callback(req, resp):
-                    logger.debug("Signal received by " + str(port))
-
-                service = RpcService(proto.EventsClientService_Stub,
-                                     port, 'localhost')
-                service.signal(request, callback=callback)
-        # send response back to client
-        response = proto.EventResponse()
-        response.status = proto.EventResponse.OK
-        done.run(response)
-
-    def ping(self, controller, request, done):
+        self._pub = self._zmq_bind(txzmq.ZmqPubConnection, port)
```

Comment:
hmm maybe it's good to leave the exception catching to the caller. I was more worried about the possibility that the port is already in use by another application.